coalib/io/FileFactory.py

```diff
@@ -82,30 +84,29 @@ def get_line(self, line):
         """
         return self.lines[line]
 
-    @property
+    @Memoize
     def lines(self):
         """
         :return:
             A tuple containing the lines of the file.
         """
         return tuple(self.string.splitlines())
 
-    @property
+    @Memoize
     def raw(self):
         """
         :return:
             The file contents as a byte sequence.
         """
-        with open(self._filename, 'rb') as fp:
-            return fp.read()
+        return Memoize.get_content(self, self._filename)
```

Comment:
Ah... don't implement `get_content` in `Memoize`, but here, and just memoize `raw` directly:
```
@Memoize
def raw(self):
    with open(...).....
```